pallets/currencies/src/fungibles.rs

```diff
@@ -79,49 +104,83 @@ where
 	<T::NativeCurrency as fungible::Inspect<T::AccountId>>::Balance: Into<BalanceOf<T>> + From<BalanceOf<T>>,
 	WithdrawConsequence<BalanceOf<T>>:
 		From<WithdrawConsequence<<T::NativeCurrency as fungible::Inspect<T::AccountId>>::Balance>>,
-	Result<Self::Balance, DispatchError>:
-		From<Result<<T::NativeCurrency as fungible::Inspect<T::AccountId>>::Balance, DispatchError>>,
-	Result<Self::Balance, DispatchError>:
-		From<Result<<T::MultiCurrency as fungibles::Inspect<T::AccountId>>::Balance, DispatchError>>,
 {
-	fn mint_into(asset: Self::AssetId, who: &T::AccountId, amount: Self::Balance) -> DispatchResult {
-		<Pallet<T>>::deposit(asset, who, amount)
+	fn handle_dust(dust: fungibles::Dust<T::AccountId, Self>) {
+		let asset = dust.0;
+		if asset == T::GetNativeCurrencyId::get() {
+			<T::NativeCurrency as fungible::Unbalanced<T::AccountId>>::handle_dust(fungible::Dust(dust.1.into()))
+		} else {
+			<T::MultiCurrency as fungibles::Unbalanced<T::AccountId>>::handle_dust(fungibles::Dust(
+				dust.0.into(),
+				dust.1.into(),
+			))
+		}
 	}
 
-	fn burn_from(
+	fn write_balance(
 		asset: Self::AssetId,
 		who: &T::AccountId,
 		amount: Self::Balance,
-	) -> Result<Self::Balance, DispatchError> {
+	) -> Result<Option<Self::Balance>, DispatchError> {
 		if asset == T::GetNativeCurrencyId::get() {
-			<T::NativeCurrency as fungible::Mutate<T::AccountId>>::burn_from(who, amount.into()).into()
+			let result = <T::NativeCurrency as fungible::Unbalanced<T::AccountId>>::write_balance(who, amount.into())?;
+			Ok(result.map(|balance| balance.into()))
 		} else {
-			<T::MultiCurrency as fungibles::Mutate<T::AccountId>>::burn_from(asset.into(), who, amount.into()).into()
+			let result = <T::MultiCurrency as fungibles::Unbalanced<T::AccountId>>::write_balance(
+				asset.into(),
+				who,
+				amount.into(),
+			)?;
+			Ok(result.map(|balance| balance.into()))
+		}
+	}
+
+	fn set_total_issuance(asset: Self::AssetId, amount: Self::Balance) {
+		if asset == T::GetNativeCurrencyId::get() {
+			<T::NativeCurrency as fungible::Unbalanced<T::AccountId>>::set_total_issuance(amount.into())
+		} else {
+			<T::MultiCurrency as fungibles::Unbalanced<T::AccountId>>::set_total_issuance(asset.into(), amount.into())
+		}
+	}
+
+	fn deactivate(asset: Self::AssetId, amount: Self::Balance) {
+		if asset == T::GetNativeCurrencyId::get() {
+			<T::NativeCurrency as fungible::Unbalanced<T::AccountId>>::deactivate(amount.into())
+		} else {
+			<T::MultiCurrency as fungibles::Unbalanced<T::AccountId>>::deactivate(asset.into(), amount.into())
+		}
+	}
+
+	fn reactivate(asset: Self::AssetId, amount: Self::Balance) {
+		if asset == T::GetNativeCurrencyId::get() {
+			<T::NativeCurrency as fungible::Unbalanced<T::AccountId>>::reactivate(amount.into())
+		} else {
+			<T::MultiCurrency as fungibles::Unbalanced<T::AccountId>>::reactivate(asset.into(), amount.into())
 		}
 	}
 }
 
-impl<T: Config> fungibles::Transfer<T::AccountId> for FungibleCurrencies<T>
```

Comment:
where did the Transfer go ? 

How can we use this now for the transfeR?

Reply:
`Transfer` trait was merged to `Mutate`. Check this [PR](https://github.com/paritytech/substrate/pull/12951) for more details. But we probably want to implement the `Mutate::transfer()` method directly.